From e9638c07e3c52949a2c65f495cd5fc74bb590040 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:12:14 -0400
Subject: [ethernet] i40e/i40evf: fix poll weight

Message-id: <1409663591-32531-92-git-send-email-sassmann@redhat.com>
Patchwork-id: 90575
O-Subject: [RHEL7.1 PATCH 091/148] i40e/i40evf: fix poll weight
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jesse Brandeburg <jesse.brandeburg@intel.com>

Fix a coding error where during the registration for NAPI
the driver requested 256 budget.  The max recommended
value for this is NAPI_POLL_WEIGHT or 64.

Change-ID: I03ea1e2934a84ff1b5d572988b18315d6d91c5c6
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit eefeacee77ecae81a7fb0dd6d710321479c48cee)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40evf_main.c b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
index e32b8c5..edb3b49 100644
--- a/drivers/net/ethernet/intel/i40evf/i40evf_main.c
+++ b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
@@ -1169,7 +1169,7 @@ static int i40evf_alloc_q_vectors(struct i40evf_adapter *adapter)
   q_vector->vsi = &adapter->vsi;
   q_vector->v_idx = q_idx;
   netif_napi_add(adapter->netdev, &q_vector->napi,
-           i40evf_napi_poll, 64);
+           i40evf_napi_poll, NAPI_POLL_WEIGHT);
   adapter->q_vector[q_idx] = q_vector;
  }
 
-- 
1.7.1